recipes/_update_shcluster.rb

```diff
@@ -0,0 +1,29 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# Recipe:: _update_shcluster
+#
+# Recipe determines whether to add/remove this SH member to/from SH Cluster.
+
+password_file = File.join node['splunk']['external_config_directory'], 'password'
+search_heads = CernerSplunk.my_cluster_data(node)['sh_cluster']
+
+fail 'Search Heads are not configured for sh clustering in the cluster databag' if search_heads.nil? || search_heads.empty?
+
+unless node['splunk']['config']['sh_member'].nil?
+  if node['splunk']['config']['sh_member'] == 'add'
+    notifies_action = :nothing
+    command = "add shcluster-member -current_member_uri #{search_heads.first}"
+  elsif node['splunk']['config']['sh_member'] == 'remove'
+    notifies_action = :stop
+    command = 'remove shcluster-member'
+  else
+    fail "node['splunk']['config']['sh_member'] accepts only values 'add' and 'remove'"
+  end
+
+  execute "#{node['splunk']['config']['sh_member']} the search head" do # ~FC037
```

Comment:
We should probably refactor the code a bit to fix the foodcritic error here rather than ignoring it.